crates/cheatable-starknet/src/rpc.rs

```diff
@@ -681,6 +685,30 @@ pub fn execute_inner_call(
     Ok(retdata_segment)
 }
 
+fn get_ret_data_by_call_entry_point<'a>(
+    call: &CallEntryPoint,
+    cheated_state: &'a CheatedState,
+) -> Option<&'a Vec<StarkFelt>> {
+    if let Some(contract_address) = call.code_address {
+        if cheated_state
+            .mocked_functions
+            .contains_key(&contract_address)
+        {
+            let contract_functions = cheated_state
+                .mocked_functions
+                .get(&contract_address)
+                .unwrap_or_else(|| {
+                    panic!("ContractAddress {contract_address:?} not found in mocked functions map")
+                });
```

Comment:
I think this could be refactored so we do not have to first check if map contains something and then get with unwrap. 

Something like `if let Some(contract_functions) = cheated_state.mocked_functions.get(&contract_address)` or a `map`.